src/main/scala/io/buoyant/linkerd/RateLimiter.scala

```diff
@@ -0,0 +1,27 @@
+package io.buoyant.linkerd
+
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.function.IntUnaryOperator
+
+import com.twitter.finagle.{Service, SimpleFilter}
+import com.twitter.finagle.http.{Request, Response, Status}
+import com.twitter.util.{Duration, Future, Timer}
+
+class RateLimiter(limit: Int, timer: Timer, intervalSeconds: Int) extends SimpleFilter[Request, Response] {
```

Comment:
This can be ever so slightly cleaner by passing in a Duration here for the interval instead of an Int.

Reply:
Good suggestions, thanks!